src/auth/jwtclient.ts

```diff
@@ -103,7 +103,7 @@ export class JWT extends OAuth2Client {
                                      target_audience: string
                                    }).target_audience) {
         const {tokens} = await this.refreshToken();
-        return {headers: {Authorization: `Bearer ${tokens.access_token}`}};
+        return {headers: {Authorization: `Bearer ${tokens.id_token}`}};
```

Comment:
I'm not familiar with IAP authentication. So now `access_token` is not used anymore? Should it be deleted from `Credentials` and `CredentialRequest` then?

Reply:
Nah.  IAP is weird.  If you pass a "target_audience" claim to the Google token endpoint, it will return an id_token instead of an access_token.  For most cases, users won't be passing that claim, except for IAP users.  We need to keep the access_token and such in both those interfaces.

Reply:
I see. A follow-up question. I don't see any usages of `access_token` in jwtclient.ts. Does it mean that JWT uses only IAP authentication?

Reply:
Nope!  JWTClient inherits from OAuth2Client, which *does* make use of the access_token that's returned in other cases.